### PR TITLE
chore(repo): remove old release scripts pt2

### PR DESCRIPTION
**Pull request by rwaskiewicz**
**Do not merge, waiting for parent to land first**
## What is the current behavior?

We have a good deal of code related to releases that no longer runs.

GitHub Issue Number: N/A

## What is the new behavior?

now that stencil releases are largely automated, release tasks that
were skipped during manual releases can be removed. this commit removes
all tasks that were skipped with `isCI` is was `true` (as it's always
`true` these days).

additional conditionals that use logical-AND with `isCI` have been
updated, as that part of the conditional is now always `true`.

remove the `--otp` flag, as it is only used for manual releases

as a result of these removals, the `--any-branch` flag can be safely
removed from our release pipeline, as its no longer used.

## Documentation

N/A - this old process being removed had no documentation 🙈 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I kicked off a dev publish of Stencil using this branch, which ran successfully (and published to NPM) https://github.com/ionic-team/stencil/actions/runs/7917891689

I was able to run this build (`4.12.2-dev.1708009206.09089a5`) locally in a basic Stencil project.

There's _some_ risk here as I haven't cut a prod release, but could cut a prerelease of our next version if someone wants to test that with me 🙂

### CODE REVIEW

File: scripts/release-tasks.ts
Comment (rwaskiewicz) [inline scripts/release-tasks.ts:10]: `postGithubRelease` is still in the codebase, but technically no longer used. Leaving it here for the moment to see if I can get GitHub to do this for us

File: scripts/release-tasks.ts
Comment (rwaskiewicz) [inline scripts/release-tasks.ts:97]: I recommend checking this section out with whitespace disabled.

File: scripts/release-tasks.ts
Comment (christian-bromann) [inline scripts/release-tasks.ts:87]: Non blocking suggestion to make it also a bit more readable using `async/await` instead of promise chaining:

```suggestion
    task: async () => {
      await execa('git', ['fetch'])
      // Retrieve the prefix for a version string - https://docs.npmjs.com/cli/v7/using-npm/config#tag-version-prefix
      const { stdout: tagPrefix } = await execa('npm', ['config', 'get', 'tag-version-prefix'])
        .catch(() => ({}))
      // verify that a tag for the new version string does not already exist by checking the output of
      // `git rev-parse --verify`
      const { stdout: hasGitTag } = await execa('git', ['rev-parse', '--quiet', '--verify', `refs/tags/${tagPrefix}${newVersion}`]).catch((err) => {
        // Command fails with code 1 and no output if the tag does not exist, even though `--quiet` is provided
        // https://github.com/sindresorhus/np/pull/73#discussion_r72385685
        if (err.stdout !== '' || err.stderr !== '') {
          throw err;
        }
      });
      if (hasGitTag) {
        throw new Error(`Git tag \`${tagPrefix}${newVersion}\` already exists.`);
      }
    },
    skip: () => isDryRun,
```

**Review by christian-bromann — APPROVED** [review]
🔪  LGTM 👍

**Review by tanner-reits — APPROVED** [review]

File: scripts/release-tasks.ts
Comment (rwaskiewicz) [reply scripts/release-tasks.ts:87]: I'm gonna hold off on this for the moment - I want to see if we can eliminate `Listr` here entirely. I started to do this (`async` work) and a few `if` statement reorders and felt like I was optimizing something that might be a better off being rewritten with some testing under it (and I didn't get that far on JEDI days 😉)

**Comment by rwaskiewicz** [comment]
I'll merge this after today's Stencil release

**Pull request merged**
